pkg/operator/finalizer.go

```diff
@@ -14,11 +14,20 @@ import (
 	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
 	regopset "github.com/openshift/client-go/imageregistry/clientset/versioned/typed/imageregistry/v1"
 	"github.com/openshift/cluster-image-registry-operator/pkg/parameters"
+	"github.com/openshift/cluster-image-registry-operator/pkg/storage"
 )
 
 func (c *Controller) RemoveResources(o *imageregistryv1.Config) error {
 	c.setStatusRemoving(o)
-	return c.generator.Remove(o)
+	err := c.generator.Remove(o)
+	switch {
+	case err == storage.ErrStorageNotConfigured:
+		return newPermanentError("StorageNotConfigured", err)
```

Comment:
This shouldn't block us from removing the resources. We certainly can remove resources that do not require the storage (service accounts, node ca deamon, etc), ideally we should be able to remove Deployment as well.

Reply:
I have refactored this PR, PTAL. I still  have to change the commit comment though but i prefer to wait until we get into an agreement.